### PR TITLE
Show last saved date

### PR DESCRIPTION
**Pull request by oicramps**
# Summary

- Adds a status bar to `logic` and `conceptual` components that show the date when the model was last saved.
- Creates a component to the `bug report button` that is shown on almost every page. Suggested by @idmarjr [here](https://github.com/brmodeloweb/brmodelo-app/pull/263#discussion_r753545244)

# Screenshots

~~https://user-images.githubusercontent.com/7519115/142931030-50724ddc-1d51-4af4-830c-c4ca5d9b7fa7.png~~

save-info-1

### CODE REVIEW

**Comment by oicramps** [comment]
@idmarjr I did the status bar as simple as possible, so the look and feel we can discuss here, and If we want to show a relative date, like `an hour ago, `now`, and so on, we'll need a library to handle this or write this logic by ourselves.

**Comment by miltonbsn** [comment]
> @idmarjr I did the status bar as simple as possible, so the look and feel we can discuss here, and If we want to show a relative date, like `an hour ago, `now`, and so on, we'll need a library to handle this or write this logic by ourselves.

To be honest I am not very comfortable with this solution, specially the idea of placing it below the canvas. I would prefer to explore another visual options before we merge it, like finding a discreet place, font and format around this unused area

image

File: app/angular/conceptual/conceptual.js
Comment (miltonbsn) [inline app/angular/conceptual/conceptual.js]: I think there is problematic merge around here

**Review by miltonbsn — CHANGES_REQUESTED** [review]
Issues I would like to discuss:

- visual place for the last saved date
- app/angular/service/logicService.js was entirely formatted (with prettier I think)
- there were a merge mistake, this pr is removing the issues #229 and pr #259

File: app/angular/conceptual/conceptual.js
Comment (oicramps) [reply app/angular/conceptual/conceptual.js]: I'll check, @miltonbsn. Thank you

File: app/angular/logic/logic.js
Comment (miltonbsn) [inline app/angular/logic/logic.js]: I believe it is another effect of a merge

**Comment by idmarjr** [comment]
Just for the records: I will take a look at this merge request in the next 5 days and explore some other visual options so we can take a decision :)

**Comment by idmarjr** [comment]
I want your opinion:
@miltonbsn @oicramps @ArthurMota9

| Option 1 🚀  | Option 2 👀  | Option 3 🎉  |
| -------- | --------- | --------- | 
| save-info-1 | save-info-2 | save-info-3 |

**Comment by ArthurMota9** [comment]
> I want your opinion: @miltonbsn @oicramps @ArthurMota9
> 
> Option 1	Option 2	Option 3

Option 3.

**Comment by oicramps** [comment]
> I want your opinion:
> @miltonbsn @oicramps @ArthurMota9
> 
> | Option 1 | Option 2 | Option 3 |
> | -------- | --------- | --------- | 
> | save-info-1 | save-info-2 | save-info-3 |
> 
>

Option 1

**Comment by idmarjr** [comment]
@oicramps I did some changes to match the visuals with the [most voted option](https://github.com/brmodeloweb/brmodelo-app/pull/266#issuecomment-985907800).

~~In this mean time I notice we're always displaying the "last save data" as the current our. Can you double check if this is the case?~~ (Fix 3c495fe)

Thanks!

**Comment by oicramps** [comment]
> @oicramps I did some changes to match the visuals with the [most voted option](https://github.com/brmodeloweb/brmodelo-app/pull/266#issuecomment-985907800).
> 
> In this mean time I notice we're always displaying the "last save data" as the current our. Can you double check if this is the case?
> 
> Thanks!

@idmarjr do you mean, when we open the model it shows the current date instead of the date that the model was saved for the last time?

**Comment by idmarjr** [comment]
> @idmarjr do you mean, when we open the model it shows the current date instead of the date that the model was saved for the last time?

That's correct!
Luckily you can overcome the terrible way I express myself. ahahah

**Review by idmarjr — APPROVED** [review]

**Comment by idmarjr** [comment]
@miltonbsn looks like this merge is blocked by a _requested change_ from you.
Can you please take a look if this is still a thing?

**Review by miltonbsn — APPROVED** [review]

**Pull request merged**
